lib/checkother.cpp

```diff
@@ -1354,6 +1354,27 @@ static std::size_t estimateSize(const Type* type, const Settings* settings, cons
 
 static bool canBeConst(const Variable *var)
 {
+	{  // Ticket #8570
+		const Function* func_scope = var->scope()->function;
+		if (func_scope->type == Function::Type::eConstructor) {
+			//could be initialized in initializer list
+			if (func_scope->arg->link()->next()->str() == ":") {
+				for (const Token* tok2 = func_scope->arg->link()->next()->next(); tok2 != var->scope()->bodyStart; tok2 = tok2->next()) {
+					if (tok2->varId() != var->declarationId())
+						continue;
+
+					const Token* parent = tok2->astParent();
+					if (parent && parent->str() == "{") //is initializer variable
+						return true;
```

Comment:
do not return true just because the first variable usage indicates the variable can be const.. all variable usages must be considered.

Reply:
My bad. I'll correct it.